### DISCUSSION TITLE
Destroying portal traps

### DISCUSSION THREAD

**Discussion by Coolthulhu** [cataclysmbn/Cataclysm-BN#3241] [General]
### Is your feature request related to a problem? Please describe.

Players commonly ask for a way to destroy the portal traps - ones that spawn enemies infinitely when in bubble.

Right now, I don't think we have a good way to do it. Portals can't be disarmed, destroying the ground doesn't seem to remove them.
The only way is to get a lucky artifact spawn - artifacts can be powered by portals, which will destroy them. You will still need to stand on the portal for up to a minute for the recharge to proc.

### Describe the solution you'd like

I see the following options:
* Make the portals not infinite - have them randomly collapse after spawning things, or spawn a boss whose death will collapse the portal
* Let player close the portal by entering it and taking a negative effect from the experience
* Make portals obsolete, remove them from all spawns and mapgens
* Remove portals from all spawns and mapgens, except for that one mega disappointing lab finale where you can use a computer to toggle a portal on and off

**Comment by chaosvolt** [comment]
I'd be in favor of going the Arcana route: making a specific JSON-defined ARTC_PORTAL item plot-relevant, and nudge the player to hunt it down for that purpose. If we ever get around to porting over the relicgen system, this effect could be added to the list of properties for relics generated in portal-themed labs.

The second option of giving the player a way to close the portals, much like with the endgame portal, at the expense of risk and consequences, would also be a good method to implement alongside this. Tying it in with the proposed idea of making our own portal dungeons for example, having some Oblivion-style pockets of liminal space held up by each portal that's just barely infected with reality enough for the player to poke around in, destabilize in some way, then book it back out with the threat of being trapped in a now-collapsed bubble of unreality if they don't make it.
